2020-02-Spring/0_arrays_matrix/README.md

```diff
@@ -119,21 +125,43 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
-
-Source: TODO :bug:
-
-#### Naive/Simple Solution
-
-TODO :bug:
-
-#### Optimal Solution
-
-TODO :bug:
-
-#### Testing The Solutions OR Driver For Solution
-
-TODO :bug:
+### 2. SOLUTION 2
+
+Source: https://leetcode.com/problems/maximum-subarray/
+
+We will traverse over the array and keep track of 2 variables: Total maximum sum, and the current sum.
+We will look at each number and add it to the current sum, if the current sum becomes negative, we ignore the numbers that made it negative.
+Resetting the current to 0 is basically saying "Ignore everything we looked at previously because it's not getting us the greatest sum."
+So if we are looking at this array: `[-2,1,-3,4,-1,2,1,-5,4]`
+We will look at -2 and see we have a negative number, so set curr to 0. 
+Then we see that 1 is postive, set curr & max to 1. 
+Then -3 gives us a curr of -2, set curr to 0, but the max is still 1.
+Then 4 gives us a curr of 4, and update max to 4. Since our curr is now 4, we have ignored all the values before 4 (-2, 1, -3).
+
+#### Solution
+
+```java
+public int maxSubArray(int[] nums) {
```

Comment:
Code style for this is good. Short, sweet, and highly readable. (I use the same bracket style so it's definitely up my alley, but I appreciate the judicious choice of variable names etc as well.) :tada: 

I traced through a few examples by hand. This solution appears to work for the edge cases I came up with, but I strongly encourage doing the following:
1. Writing tests to prove it works.
2. Tracing through edge cases with this code during the session (or even giving example input to let folks test their own solutions by and or in an editor).

Here are a few edge cases I tried:
* All negative numbers (`[ -3 -16 -4 -1 -2 -6]`)
* Not front-loaded arrays, where the max is at the end of the array (`[ -1 1 3 4 5 ]`)
* A normal-ish array (`[ 3 2 6 0 1 -3 1 ]`)

Reply:
Will add those, thank you!